.github/actions/build-publish-chainlink/action.yml

```diff
@@ -127,3 +141,22 @@ runs:
       build-args: |
         CHAINLINK_USER:chainlink
         ${{ env.shared-build-args }}
+
+  - if: inputs.sign-images == 'true'
+    name: Install cosign
+    uses: sigstore/cosign-installer@1e95c1de343b5b0c23352d6417ee3e48d5bcd422 # v1.4.0
+    with:
+      cosign-release: 'v1.4.0'
+
+  - if: inputs.sign-images == 'true'
+    name: Write signing key to disk (only needed for `cosign sign --key`)
+    shell: sh
+    run: echo "${{ inputs.cosign-private-key }}" > cosign.key
+
+  - if: inputs.sign-images == 'true'
+    name: Sign the published Docker image
+    shell: sh
+    env:
+      COSIGN_PASSWORD: "${{ inputs.cosign-password }}"
+    run: cosign sign --key cosign.key ${{ inputs.ecr-registry }}/${{ inputs.image-name }}@${{ steps.meta-nonroot.outputs.digest }}
```

Comment:
Do we only want to sign the non-root digest here? Seems like we want to sign both generated images, no?

Reply:
Also, the metadata action doesnt output the image digest, the build-push-action does. Another output the build-push-action provides is `metadata`, you might want to take a look in there and see if it makes forming the image uri easier